libs/muddle/src/peer_selector.cpp

```diff
@@ -358,10 +359,20 @@ void PeerSelector::OnAnnouncement(Address const &from, byte_array::ConstByteArra
   static constexpr auto CACHE_LIFETIME = MAX_ANNOUNCEMENT_INTERVAL + MIN_ANNOUNCEMENT_INTERVAL;
   FETCH_UNUSED(payload);
 
-  Peers peers{};
+  // load the peer list from the network
+  Peers peer_list{};
+  try
+  {
+    serializers::MsgPackSerializer serializer{payload};
+    serializer >> peer_list;
+  }
+  catch (std::exception const &ex)
+  {
+    FETCH_LOG_ERROR(logging_name_, "Unable to deserialise announcement packet: ", ex.what());
+  }
 
-  serializers::MsgPackSerializer serializer{payload};
-  serializer >> peers;
+  // convert to a set
+  std::unordered_set<network::Peer> const peers{peer_list.begin(), peer_list.end()};
```

Comment:
You are correct, the issue here is that the serialisation object is a `std::vector` here. While it is true that the current implementation of the serialisation for `std::vector` and `std::unordered_set` is the same (it is just a message pack array). There is a chance that this will not be the case in the future. This is why I elected to make this copy.

To be clear in all use cases at the moment this is a vector of size 1, so while the addition copy is not great, the overhead is marginal.

@troelsfr is planning a big update in this area, I suspect that this code will be completely replaced. It would be therefore my preference to keep this inefficiency for the time being.